Sources/Controllers/2.0/OAMapSettingsWikipediaLanguagesScreen.mm

```diff
@@ -0,0 +1,428 @@
+{
+    NSDictionary *item = [self getItem:indexPath];
+    if ([item[@"type"] isEqualToString:[OADividerCell getCellIdentifier]])
+    {
+        OADividerCell* cell = [tableView dequeueReusableCellWithIdentifier:[OADividerCell getCellIdentifier]];
+        if (cell == nil)
+        {
+            NSArray *nib = [[NSBundle mainBundle] loadNibNamed:[OADividerCell getCellIdentifier] owner:self options:nil];
+            cell = (OADividerCell *) nib[0];
+            cell.backgroundColor = UIColor.whiteColor;
+            cell.dividerColor = UIColorFromRGB(color_tint_gray);
+            cell.dividerInsets = UIEdgeInsetsZero;
+            cell.dividerHight = 0.5;
+        }
+        return cell;
+    }
+    else if ([item[@"type"] isEqualToString:[OASettingSwitchCell getCellIdentifier]])
+    {
+        OASettingSwitchCell* cell = [tableView dequeueReusableCellWithIdentifier:[OASettingSwitchCell getCellIdentifier]];
+        if (cell == nil)
+        {
+            NSArray *nib = [[NSBundle mainBundle] loadNibNamed:[OASettingSwitchCell getCellIdentifier] owner:self options:nil];
+            cell = (OASettingSwitchCell *) nib[0];
+        }
+
+        if (cell)
+        {
+            cell.separatorInset = UIEdgeInsetsMake(0.0, 0.0, 0.0, 0.0);
+            cell.textView.text = item[@"title"];
+            cell.descriptionView.hidden = YES;
+            cell.imgView.hidden = YES;
+            [cell.switchView setOn:_isGlobalWikiPoiEnabled];
+            cell.selectionStyle = UITableViewCellSelectionStyleNone;
+            cell.switchView.tag = indexPath.section << 10 | indexPath.row;
+            [cell.switchView removeTarget:nil action:NULL forControlEvents:UIControlEventValueChanged];
+            [cell.switchView addTarget:self action:@selector(applyParameter:) forControlEvents:UIControlEventValueChanged];
+        }
+        return cell;
+    }
+    else if ([item[@"type"] isEqualToString:[OAMenuSimpleCellNoIcon getCellIdentifier]])
+    {
+        OAWikiLanguageItem *language = item[@"item"];
+        OAMenuSimpleCellNoIcon *cell = [tableView dequeueReusableCellWithIdentifier:[OAMenuSimpleCellNoIcon getCellIdentifier]];
+        if (cell == nil)
+        {
+            NSArray *nib = [[NSBundle mainBundle] loadNibNamed:[OAMenuSimpleCellNoIcon getCellIdentifier] owner:self options:nil];
+            cell = (OAMenuSimpleCellNoIcon *) nib[0];
+            cell.tintColor = UIColorFromRGB(color_primary_purple);
+            UIView *bgColorView = [[UIView alloc] init];
+            bgColorView.backgroundColor = [UIColor colorWithWhite:1.0 alpha:0.0];
+            [cell setSelectedBackgroundView:bgColorView];
+        }
+        if (cell)
+        {
+            cell.separatorInset = UIEdgeInsetsMake(0.0, 62.0, 0.0, 0.0);
+            cell.textView.text = [OAUtilities capitalizeFirstLetterAndLowercase:language.title];
+            cell.descriptionView.hidden = YES;
```

Comment:
This can be done once during cell creation (incl. if ([cell needsUpdateConstraints])
                [cell updateConstraints];)